src/Modules/Render/ViewScene.cc

```diff
@@ -233,6 +234,9 @@ void ViewScene::syncMeshComponentFlags(const std::string& connectedModuleId, Mod
 
 void ViewScene::execute()
 {
+  auto state = get_state();
+  state->setValue(IsExecuting, true);
```

Comment:
This should change to a scoped object--set state value to true in constructor, then set to false in destructor which ensures it gets called on exit of this function. Since execute might return early, or throw, with such an object you don't need to worry about multiple setValue(false) calls scattered around. See `ag Scoped` for examples.

Reply:
If I create a scoped class, is there a way to save it to state so that the interface can see it?